### ISSUE TITLE
[SCREENREADER]  Wrap ErrorableDate component in a fieldset

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#15364]
**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Point of Contact

**VFS Point of Contact:** _First name only. Delete if not applicable._

## User Story or Problem Statement

The `ErrorableDate` component is currently wrapped in a `div` with a `label` inside. To ensure the month, day and year fields are all associated, the component should be wrapped in a `fieldset` with associated `legend`.

## Details

Currently, this problem can be seen in the form 526 wizard - https://staging.va.gov/disability/file-disability-claim-form-21-526ez/introduction - 

![Screen Shot 2020-10-27 at 1 02 39 PM](https://user-images.githubusercontent.com/136959/97344492-d2556d00-1856-11eb-8f4f-fbf209f58399.png)

The two `ErrorableRadioButtons` components immediately above the `ErrorableDate` _are_ each wrapped in a `fieldset`, as seen in the screenshot of the dev tools inspector.

## Acceptance Criteria

- [ ] Errorable date components are all wrapped in a `fieldset` with associated `legend`
- [ ] The `label` property should be marked as `required` as a [`fieldset` _must_ include a non-empty `legend`](https://codepen.io/stevef/post/fielsdet-and-legend-semantics).

## Definition of done

1. `ErrorableDate` component is wrapped appropriately
1. Update `ErrorableDate` documentation.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Environment

All

## Steps to Recreate

1. No need to log in
1. Go to https://staging.va.gov/disability/file-disability-claim-form-21-526ez/introduction
1. In the wizard, choose "No" for the active duty question
1. Choose "I'm disagreeing with a VA decision..."
1. Examine the now visible `ErrorableDate` component

## Solution (if known)

[Code changes around line 87](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/master/packages/formation-react/src/components/ErrorableDate/ErrorableDate.jsx#L87)

```diff
     return (
-      <div className={!isValid ? 'input-error-date' : undefined}>
-        <label>
+      <fieldset className={!isValid ? 'input-error-date' : ''}>
+        <legend>
           {this.props.label ? this.props.label : 'Date of birth'}
           {this.props.required && (
             <span className="form-required-span">(*Required)</span>
           )}
-        </label>
+        </legend>



-      </div>
+      </fieldset>
     );
```

**Comment by meganhkelley** [comment]
Is this forms library? If so the new point of contact is @caw310 for the VSP Design System team! Thanks

**Comment by Mottie** [comment]
Yes! Sorry, I'll get it right next time!
